src/effects/builtin/truncatesilence/TruncateSilenceView.qml

```diff
@@ -0,0 +1,223 @@
+/*
+* Audacity: A Digital Audio Editor
+*/
+import QtQuick 2.15
+import QtQuick.Layouts
+import Muse.Ui 1.0
+import Muse.UiComponents 1.0
+import Audacity.Effects
+import Audacity.BuiltinEffects
+
+BuiltinEffectBase {
+    id: root
+
+    property string title: truncateSilence.effectTitle
+    property bool isApplyAllowed: true
+
+    implicitHeight: mainColumn.height // see with EffectsViewerDialog.qml
+
+    model: truncateSilence
+
+    TruncateSilenceViewModel {
+        id: truncateSilence
+    }
+
+    QtObject {
+        id: prv
+
+        readonly property int spacingS: 4
+        readonly property int spacingM: 8
+        readonly property int spacingL: 16
+
+        readonly property int fieldWidth: 156
+        readonly property int desiredWidth: 360
+
+        readonly property int borderWidth: 1
+        readonly property int borderRadius: 4
+    }
+
+    width: prv.desiredWidth - (2 * prv.spacingL) // we need to remove the padding from the dialog desired width
+
+    ColumnLayout {
+        id: mainColumn
+
+        width: parent.width
+        spacing: prv.spacingL
+
+        // Detect silence section
+        Column {
+            Layout.fillWidth: true
+            spacing: prv.spacingM
+
+            StyledTextLabel {
+                text: truncateSilence.detectSilenceLabel
+                font: ui.theme.bodyFont
+            }
+
+            RoundedRectangle {
+                width: parent.width
+                height: detectSilenceRow.height + prv.spacingL * 2
+
+                color: ui.theme.backgroundSecondaryColor
+
+                border.color: ui.theme.strokeColor
+                border.width: prv.borderWidth
+
+                radius: prv.borderRadius
+
+                Row {
+                    id: detectSilenceRow
+                    anchors.centerIn: parent
+                    width: parent.width - prv.spacingL * 2
+                    spacing: prv.spacingL
+
+                    Column {
+                        width: (parent.width - parent.spacing) / 2
+                        spacing: prv.spacingM
+
+                        StyledTextLabel {
+                            text: truncateSilence.thresholdLabel
+                        }
+
+                        TextInputField {
```

Comment:
It is better to use IncrementalPropertyControl here, as with other effect views. And again as a general rule we should never do this: 
```
currentText: value + " " + some text
```